docs/data-source-matrix.md

```diff
@@ -0,0 +1,54 @@
+# Data Source Matrices
+
+The Carbon Aware SDK includes access to various data sources of carbon aware data, including WattTime, ElectricityMaps, and a custom JSON source. These matrices are an attempt to track what features of the Carbon Aware SDK are enabled for which data sources.
+
+## Contents
+
+- [Type of Data Sources and Configuration](#type-of-data-sources-and-configuration)
+- [Data Source Routes Available](#data-source-routes-available)
+- [Location Source Availability](#location-source-availability)
+
+## Type of Data Sources and Configuration
+
+In the CarbonAware SDK configuration, you can set what data source to use as the `EmissionsDataSource` and the `ForecastDataSource`. There are also certain configuration fields that must be set in order to access the raw data.
+| Type      | WattTime  | ElectricityMaps | JSON |
+|--------------------------|-----------|-----------------|------|
+| Is Emissions DataSource   | Yes      | Yes             | Yes     |
+| Is Forecast DataSource   | Yes  | Yes            | No     |
+| Makes HTTP(s) call   | Yes  | Yes            | No    |
+| Can Use Custom Data   | No  | No            | Yes    |
+| Needs Authentication Config   | Yes - username/password | Yes - token/header            | No    |
+| BaseUrl Override Config   | Only for testing  | Switch between trial + full version, testing            | No   |
+| Supports Trial + Full Account   | Yes  | Yes (*different URL and token header required)            | N/A    |
+
+## Data Source Routes Available
```

Comment:
I found this section pretty hard to parse.  Consider removing the list and imagining the entire thing as a table.  Potentially:

| Data Source Method | WattTime  | ElectricityMaps | JSON | CLI Usage | WebAPI Usage | SDK Usage|
|--------------|:-----------:|:-----------------:|:------:|:------:|:------:|:------:|
| `GetCarbonIntensityAsync` | ✅   | ✅ | ✅ | `emissions` | `emissions/bylocation` / `emissions/bylocations` / `emissions/bylocations/best` / `emissions/average-carbon-intensity` / `average-carbon-intensity/batch` | `GetEmissionsDataAsync(...)` / `GetBestEmissionsDataAsync(...)` / `GetAverageCarbonIntensityDataAsync(...)` |